py-polars/src/lazy/dsl.rs

```diff
@@ -562,28 +562,33 @@ impl PyExpr {
         exact: bool,
         cache: bool,
         tz_aware: bool,
+        tu: Option<Wrap<TimeUnit>>,
     ) -> PyExpr {
-        let tu = match fmt {
-            Some(ref fmt) => {
-                if fmt.contains("%.9f")
-                    || fmt.contains("%9f")
-                    || fmt.contains("%f")
-                    || fmt.contains("%.f")
-                {
-                    TimeUnit::Nanoseconds
-                } else if fmt.contains("%.3f") || fmt.contains("%3f") {
-                    TimeUnit::Milliseconds
-                } else {
-                    TimeUnit::Microseconds
+        let result_tu = if tu.is_none() {
+            match fmt {
+                Some(ref fmt) => {
+                    if fmt.contains("%.9f")
+                        || fmt.contains("%9f")
+                        || fmt.contains("%f")
+                        || fmt.contains("%.f")
+                    {
+                        TimeUnit::Nanoseconds
+                    } else if fmt.contains("%.3f") || fmt.contains("%3f") {
+                        TimeUnit::Milliseconds
+                    } else {
+                        TimeUnit::Microseconds
+                    }
                 }
+                None => TimeUnit::Microseconds,
             }
-            None => TimeUnit::Microseconds,
+        } else {
+            tu.unwrap().0
```

Comment:
I couldn't figure out how to do this

If I write `tu.0`, then `make build` gives:
```
   Compiling py-polars v0.15.14 (/home/marcogorelli/polars-dev/py-polars)
error[E0609]: no field `0` on type `std::option::Option<conversion::Wrap<polars::prelude::TimeUnit>>`                                          
   --> src/lazy/dsl.rs:585:16
    |
585 |             tu.0
    |                ^


error: aborting due to previous error> ] 334/335: py-polars                                                                                    


For more information about this error, try `rustc --explain E0609`.

error: could not compile `py-polars` due to 2 previous errors
💥 maturin failed
  Caused by: Failed to build a native library through cargo
  Caused by: Cargo build finished with "exit status: 101": `"cargo" "rustc" "--message-format" "json" "--lib"`
```

On the other hand, if I put `tu.unwrap().0`, then `make build` works fine, but `make pre-commit` gives:

```
    Checking py-polars v0.15.14 (/home/marcogorelli/polars-dev/py-polars)
error: called `unwrap` on `tu` after checking its variant with `is_none`
   --> src/lazy/dsl.rs:585:13
    |
567 |         let result_tu = if tu.is_none() {
    |                         --------------- help: try: `if let Some(..) = tu`
...
585 |             tu.unwrap().0
    |             ^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap
    = note: `-D clippy::unnecessary-unwrap` implied by `-D warnings`

error: could not compile `py-polars` due to previous error
make: *** [Makefile:38: clippy] Error 101
```

Looking into this now (apologies, I'm new to rust!)